modules/nw-creating-secured-routes.adoc

```diff
@@ -0,0 +1,94 @@
+// Module filename: nw-creating-secured-routes.adoc
+// Module included in the following assemblies:
+// * networking/configuring-routing.adoc
+
+[id='nw-creating-secure-routes-{context}']
+= Creating Secured Edge Terminated Routes
+
+You can create a secure edge terminated route where TLS
+termination occurs on the router before the traffic is proxied to the
+destination.
+The secure edge terminated route specifies the TLS certificate
+and key information.
+The TLS certificate is served by the router front end.
+
+.Procedure
+. Create a private key for the edge secured route.
++
+----
+# sudo openssl genrsa -out example-test.key 2048
+----
++
+[NOTE]
+====
+Currently, password protected key files are not supported. HAProxy prompts
+for a password upon starting and does not have a way to automate this process.
+To remove a passphrase from a keyfile, use the following command:
+----
+# openssl rsa -in <passwordProtectedKey.key> -out <new.key>
+----
+====
+
+. Create a CSR for the edge secured route.
++
+----
+# sudo openssl req -new -key example-test.key -out example-test.csr  \
+  -subj "/C=US/ST=CA/L=Mountain View/O=OS3/OU=Eng/CN=www.example.test"
+----
+
+. Create a certificate for the edge secured route.
++
+The procedure to create certificates is specific to your certificate
+authority and provider.
+The following command generates a simple self-signed certificate for a
+domain named `www.example.test`,
++
+----
+# sudo openssl x509 -req -days 366 -in example-test.csr  \
+      -signkey example-test.key -out example-test.crt
+----
+
+. Generate a route using the certificate and key that you created.
++
+----
+$ oc create route edge --service=my-service \
+    --hostname=www.example.test \
+    --key=example-test.key --cert=example-test.crt
+route "my-service" created
+----
+
+. View the route definition.
++
+----
+$ oc get route/my-service -o yaml
+apiVersion: v1
+kind: Route
+metadata:
+  name:  my-service
+spec:
+  host: www.example.test
+  to:
+    kind: Service
+    name: my-service
+  tls:
+    termination: edge
+    key: |
+      -----BEGIN PRIVATE KEY-----
+      [...]
+      -----END PRIVATE KEY-----
+    certificate: |
+      -----BEGIN CERTIFICATE-----
+      [...]
+      -----END CERTIFICATE-----
+----
+
+. Verify that the DNS entry for `www.example.test` points to the router
+instance(s) and that the route to your domain is available.
++
+The example below uses curl and a local resolver to simulate the
+DNS lookup:
+
+----
+# routerip="4.1.1.1"  #  replace with IP address of one of your router instances.
+# curl -k --resolve www.example.test:443:$routerip https://www.example.test/
```

Comment:
Do these commands really need to be run as root, or can both of these "#" prompts be "$"?